crypto/kyber/pqcrystals_kyber_ref_common/poly.c

```diff
@@ -164,16 +166,17 @@ void poly_frombytes(poly *r, const uint8_t a[KYBER_POLYBYTES])
 void poly_frommsg(poly *r, const uint8_t msg[KYBER_INDCPA_MSGBYTES])
 {
   unsigned int i,j;
-  int16_t mask;
+  crypto_word_t mask;
 
 #if (KYBER_INDCPA_MSGBYTES != KYBER_N/8)
 #error "KYBER_INDCPA_MSGBYTES must be equal to KYBER_N/8 bytes!"
 #endif
 
   for(i=0;i<KYBER_N/8;i++) {
     for(j=0;j<8;j++) {
-      mask = -(int16_t)((msg[i] >> j)&1);
-      r->coeffs[8*i+j] = mask & ((KYBER_Q+1)/2);
+      mask = constant_time_lt_w(0, (msg[i] >> j) & 1);
```

Comment:
can we please use `constant_time_is_zero_w` since it makes more sense in this context?

Reply:
Sure, but we end up needing to introduce some inversion logic external to `constant_time_*`. i.e. We have to either invert the output mask or we invert the input bit. I don't think it adds risk to our compiler output, but just calling it out.